walk_test.go

```diff
@@ -552,6 +552,25 @@ func TestWalkHoltWinters(t *testing.T) {
 	require.True(t, s2.Val > 0 && s2.Val < 1)
 }
 
+func TestInfo(t *testing.T) {
+	rnd := rand.New(rand.NewSource(time.Now().Unix()))
+	opts := []Option{WithEnableOffset(true), WithEnableAtModifier(true)}
+	p := New(rnd, testSeriesSet, opts...)
+	runs := 100
+	for i := 0; i < runs; i++ {
+		expr := &parser.Call{
+			Func: parser.Functions["info"],
+			Args: make([]parser.Expr, len(parser.Functions["info"].ArgTypes)),
+		}
+
+		p.walkInfo(expr)
+		require.Equal(t, parser.ValueTypeVector, expr.Args[0].Type())
+		if len(expr.Args) == 2 {
+			require.Equal(t, parser.ValueTypeVector, expr.Args[1].Type())
```

Comment:
`expr.Args[1].Type()` only makes sure the response type is `vector`.
For this funciton, we need to make sure that `expr.Args[1]` itself is a vector selector.

Reply:
That's right. What you did is exactly what I want as this is what upstream Promql engine uses

```
_, ok := expr.Args[1].(*parser.VectorSelector)
```